From ba53c0bcbd5aeb64b3cc4ae33073d9d8c2ff07f0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michal=20=C4=8Ciha=C5=99?= <michal@cihar.com>
Date: Fri, 6 Aug 2021 14:05:30 +0200
Subject: [PATCH] variants: Avoid recursive update of variants

There is no need to do other updates to the unit model here.
---
 weblate/trans/models/component.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/weblate/trans/models/component.py b/weblate/trans/models/component.py
index 2406d7da1e60..667e05b2c487 100644
--- weblate/trans/models/component.py
+++ weblate/trans/models/component.py
@@ -2617,7 +2617,7 @@ def update_variants(self):
                     unit.variant = Variant.objects.get_or_create(
                         key=key, component=self, variant_regex=self.variant_regex
                     )[0]
-                    unit.save(update_fields=["variant"])
+                    unit.save(update_fields=["variant"], only_save=True)
 
         # Update variant links
         for variant in Variant.objects.filter(component=self).iterator():
